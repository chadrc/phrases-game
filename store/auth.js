export const state = () => ({
  signingUp: false,
  signingIn: false,
  sendingSignUp: false,
  signUpResult: null,
  signUpError: null
});

export const getters = {
  signingUp: state => state.signingUp,
  signingIn: state => state.signingIn,
  sendingSignUp: state => state.sendingSignUp,
  signUpResult: state => state.signUpResult,
  signUpError: state => state.signUpError
};

export const mutations = {
  setSigningUp: (state, val) => {
    state.signingUp = val;
  },
  setSigningIn: (state, val) => {
    state.signingIn = val;
  },
  setSendingSignUp: (state, val) => {
    state.sendingSignUp = val;
  },
  setSignUp: (state, result) => {
    if (result.error) {
      state.signUpError = result;
    } else {
      state.signUpResult = result;
    }
  }
};

export const actions = {
  startSignUp: ({ commit }) => {
    commit("setSigningUp", true);
  },
  endSignUp: ({ commit }) => {
    commit("setSigningUp", false);
  },
  startSignIn: ({ commit }) => {
    commit("setSigningIn", true);
  },
  endSignIn: ({ commit }) => {
    commit("setSigningIn", false);
  }
};
