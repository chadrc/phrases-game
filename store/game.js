import GameService from "@/services/GameService";

export const state = () => ({
  sendingStartGame: false,
  startGameError: null,
  currentGame: null
});

export const getters = {
  sendingStartGame: (state) => state.sendingStartGame,
  startGameError: (state) => state.startGameError,
  currentGame: (state) => state.currentGame
};

export const mutations = {
  setSendingStartGame: (state, val) => {
    state.sendingStartGame = val;
  },
  setStartGameError: (state, error) => {
    state.startGameError = error;
  },
  setCurrentGame: (state, game) => {
    state.currentGame = game;
  }
};

export const makeActions = gameService => {
  return {
    startGame: async ({ commit }) => {
      commit("setSendingStartGame", true);

      const startGameResponse = await gameService.startGame();

      if (startGameResponse.error) {
        commit("setStartGameError", startGameResponse);
      } else {
        commit("setCurrentGame", startGameResponse);
      }

      commit("setSendingStartGame", false);
    },
  };
};

export const actions = makeActions(new GameService());
