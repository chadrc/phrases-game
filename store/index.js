export const state = () => {
  return {
    phrases: []
  };
};

export const mutations = {
  setPhrases: (state, { phrases }) => {
    state.phrases = phrases;
  }
};
