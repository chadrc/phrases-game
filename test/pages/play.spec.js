import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import PlayView from "@/pages/play.vue";
import * as authStoreModule from "@/store/auth";

describe("Play View", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);
  const store = new Vuex.Store({
    modules: {
      auth: {
        namespaced: true,
        ...authStoreModule
      }
    }
  });

  const wrapper = mount(PlayView, {
    localVue,
    store
  });

  test("is vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders regular", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
