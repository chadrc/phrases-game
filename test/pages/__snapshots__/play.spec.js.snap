// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play View renders regular 1`] = `
<section
  class="container"
>
  <h1
    class="title has-text-centered"
  >
    
    Play
  
  </h1>
   
  <section
    class="game-area"
  >
    <button
      class="button is-primary is-large is-fullwidth"
    >
      
      New Game
    
    </button>
  </section>
</section>
`;
